src/server/routes/callback.js

```diff
@@ -110,34 +111,34 @@ export default async function callback (req, res) {
         // e.g. option to send users to a new account landing page on initial login
         // Note that the callback URL is preserved, so the journey can still be resumed
         if (isNewUser && pages.newUser) {
-          return res.redirect(`${pages.newUser}${pages.newUser.includes('?') ? '&' : '?'}callbackUrl=${encodeURIComponent(callbackUrl)}`)
+          return res.redirect(`${locale ? `/${locale}` : ''}${pages.newUser}${pages.newUser.includes('?') ? '&' : '?'}callbackUrl=${encodeURIComponent(callbackUrl)}`)
```

Comment:
OMG... 😅
Maybe we should make `res.redirect` smarter to take something like:

```res.redirect(`/${locale ?? ""}${pages.newUser}`, { callbackUrl } )```

And merge query params and a normalized (remove duplicate "/" for example. Node has [`path.normalize`](https://nodejs.org/api/path.html#path_path_normalize_path)) path to the correct URL.

What do you think?

Reply:
I'm not that familiar (yet) with the whole codebase, but I assume you mean extending .redirect like its done for other res.* methods in `extend-req.js`? (which has a misleading name by the way :p)

But yeah this isn't the only instance of url-forming spaghetti that is made worse by including locale.

Reply:
You brought my attention to this, so I did fix `extend-req`->`extend-res`: https://github.com/nextauthjs/next-auth/commit/dbe283f0fa5b0521b6a4be1f63bf916891c0d5fe  But I DID write it wrong in the commit message... 🤦 😅